test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs

```diff
@@ -605,6 +605,38 @@ public void HubsCannotHaveOverloadedMethods()
             }
         }
 
+        [Fact]
+        public async Task DelayedSendTest()
+        {
+            var serviceProvider = CreateServiceProvider();
+
+            dynamic endPoint = serviceProvider.GetService(GetEndPointType(typeof(HubT)));
```

Comment:
@Mardoxx - `serviceProvider.GetService(Type)` returns `object`. We want to call `.OnConnectedAsync` on endPoint but this is an `object` so you would have to cast it before. However we don't have a type to cast to because the type passed to `GetService` is [created](https://github.com/aspnet/SignalR/blob/ef1d5ba8c3e9ddbc16827c147fb961a9b03f8621/test/Microsoft.AspNetCore.SignalR.Tests/HubEndpointTests.cs#L770-L774) at runtime. So, we enforce `endPoint` to be `dynamic` so that we don't deal with Reflection (`GetMethod`/`Invoke`) ourselves - it's not beautiful but serves the purpose and makes the code much more readable.